autosklearn/pipeline/components/data_preprocessing/text_encoding/bag_of_word_encoding.py

```diff
@@ -44,8 +46,10 @@ def fit(self, X: PIPELINE_DATA_DTYPE, y: Optional[PIPELINE_DATA_DTYPE] = None
                                                     ngram_range=(1, self.ngram_range))
             else:
                 raise KeyError()
-            for feature in X.columns:
-                self.preprocessor = self.preprocessor.fit(X[feature])
+
+            all_text = itertools.chain.from_iterable(X[col] for col in X.columns)
+            self.preprocessor = self.preprocessor.fit(all_text)
```

Comment:
This appears to be a neat solution. Can it handle NaNs in the text?

Reply:
Shouldn't ignoring or empty string be the same? In any case, this will create a copy. Can we maybe get around not creating a copy by using functools.filter?

Reply:
```python
x = pd.Series(["hello", None, "World"])
x.dropna()  # ["hello", "world"])
x.fillna("")  # ["hello", "", "world"])
```

The CountVectorizer doesn't even know there was an Na when we drop where as with the second one, there is some value, `""` which indicates a field was blank, which can then be counted.

You're right that this does a copy, however the code to loop over all columns and remove anything that is a pandas Nan is not so pleasant.

```python
all _text = iter(
    word for word in chain.from_iterable(X[col] for col in X.columns)
    if not isinstance(word, pd._libs.missing.NAType))
)
```

Reply:
Sorry, I rather meant something along the lines of
```
all_text = filter(
    lambda x: not pd.isnull(x),
    itertools.chain.from_iterable(X[col] for col in X.columns if X[col].dtype == 'string')
)
```
but it is slower than a copy, so never mind. Let's go with your dropping solution then.

Reply:
I did some testing as well and the `dropna` way was fastest. I suppose we just drop NA's for now but take a mental not that this could be changed.
```python
all_text = itertools.chain.from_iterable(X[col].dropna() for col in X.columns)
```

```python
In [96]: def iterway(X):
    ...:     all_text = list(
    ...:         word
    ...:         for word in itertools.chain.from_iterable(X[col] for col in X.columns)
    ...:         if not isinstance(word, pd._libs.missing.NAType)
    ...:     )
    ...:     return all_text
    
In [96]: def drop_col_way(X):
    ...:     return list(itertools.chain.from_iterable(X[col].dropna() for col in X.columns))
    
In [96]: def iterway_with_pdnull(X):
    ...:     all_text = list(
    ...:         word
    ...:         for word in itertools.chain.from_iterable(X[col] for col in X.columns)
    ...:         if pd.isnull(word)
    ...:     )
    ...:     return all_text
 
In [94]: %timeit -n 1000 drop_col_way(X)
2.42 ms ± 9.14 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [95]: %timeit -n 1000 iterway(X)
3.14 ms ± 53.9 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)

In [97]: %timeit -n 1000 iterway_with_pdnull(X)
4.11 ms ± 150 µs per loop (mean ± std. dev. of 7 runs, 1,000 loops each)
```